changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,203 @@
+We do not want to support these cases.
+Users should not be setting a `max plaintext length` value for known-length plaintexts
+with the intention of changing CMM behavior.
+Instead, users should be specifying this value in for unknown-length plaintexts in order to use CMMs
+that depend on plaintext length for their internal logic.
+
+This change additionally renames the `plaintext length` field on GetEncryptionMaterialsRequest to
+`max plaintext length` because that better explains the intent behind this field.
+This field describes the max size of the plaintext which will be encrypted using the materials
+the CMM is to generate.
+This is the current state of the behavior, so the specification should describe it as such.
+This is renamed to a different name than `plaintext length bound` in order to
+distinguish the two controls.
+
+This change does not strictly prescribe how users should express intent for the plaintext
+and a possible `plaintext length bound` through an API.
+How an API should be designed with these controls in mind depends greatly on the capabilities and
+idioms of the language in use.
+Specifically, this gets tricky for languages where the idiom is to
+accept an object represent a set of user supplied params.
+If a user supplies both a known-length plaintext and a `max plaintext length`,
+what should be done?
+It doesn't make sense to have to check for bad values in the user supplied params,
+given it's unstructured nature.
+Thus, our specification reccomends that languages SHOULD either ensure that `max plaintext length`
+isn't supplied with a known-length plaintext by construction.
+If it doesn't, then it SHOULD ignore any user supplied `max plaintext length` in the case that
+a known-length plaintext is supplied.
+
+## Drawbacks
+
+This change does not prescribe the name of this input in implementations,
+and thus there will not be a standardized name or way of specifying this input across implementations.
+This is because this control already exists in implementations under different names,
+and it is not worth breaking all customers to rename this input
+without significantly changing this input to benefit the customer.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any Python user that depends on the source_length being sent to the
+CMM instead of the true length of a known-length plaintext.
+
+## Guide-level Explanation
+
+We need to update the spec to make the purpose and behavior of the `plaintext length` input clear
+by renaming it and specifying its exact behavior within the Encrypt operation:
+
+When performing the Encrypt operation on an unknown-length plaintext,
+users MUST be able to specify an optional parameter `plaintext length bound`.
+The value of this field represents the max length of the plaintext to be encrypted.
+The ESDK MUST NOT encrypt a plaintext greater than this length,
```

Comment:
I feel like this is a little buried.  I can see why it is not in the list below, but I think that a little higher billing is in order.

Reply:
Where do you think this should be highlighted more? In the summary? Reference it again later?